lib/AST/Type.cpp

```diff
@@ -759,6 +759,18 @@ bool TypeBase::isStdlibType() {
   return false;
 }
 
+bool TypeBase::isStdlibCollectionType() {
+  if (auto *structType = getAs<BoundGenericStructType>()) {
+    auto &ctx = getASTContext();
+    auto *decl = structType->getDecl();
+    if (decl == ctx.getArrayDecl() || decl == ctx.getDictionaryDecl() ||
+        decl == ctx.getSetDecl())
+      return true;
```

Comment:
Nit: You could do `return decl == ctx.getArrayDecl() || decl == ctx.getDictionaryDecl() || decl == ctx.getSetDecl()`